### PR TITLE
Changes error message in `import_app`

### LINKED ISSUE #1647: Make "failed to find application" a bit more informative
When issuing `gunicorn myapp:app` and obtaining result `failed to find application myapp`, it would be better if it was clear whether there was a problem with `myapp` or with `app`.

### PR DESCRIPTION
**Pull request by JJ**
Closes #1647 when accepted.
Adds tests for `add_import` using the existing `test`app.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,3 +14,5 @@ examples/frameworks/pylonstest/PasteScript*
 examples/frameworks/pylonstest/pylonstest.egg-info/
 examples/frameworks/django/testing/testdb.sql
 .tox
+.cache/
```
Comment (berkerpeksag) [inline .gitignore]: We usually reject changes to `.gitignore` because it's better to use a [global `.gitignore`](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore) for user-specific directories.

File: gunicorn/util.py [hunk 2]
```diff
@@ -363,7 +363,7 @@ def import_app(module):
     except NameError:
         if is_debug:
             traceback.print_exception(*sys.exc_info())
-        raise AppImportError("Failed to find application: %r" % module)
+        raise AppImportError("Failed to find application object %r in %r" % (obj,module) )
```
Comment (berkerpeksag) [inline gunicorn/util.py]: Style nits: Please add a space after `,` and remove the space between the two closing parens.

File: tests/test_util.py [hunk 3]
```diff
@@ -6,7 +6,16 @@
...
+from gunicorn.errors import AppImportError
+import sys
+import os
 
+# Obtain project root path
+project_root = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+# For finding modules to test
+sys.path.insert(0, project_root)
+sys.path.insert(0, project_root + '/examples')
```
Comment (berkerpeksag) [inline tests/test_util.py]: We are going to need to create some apps for testing sooner or later so I say let's use this PR an opportunity to create a basic WSGI app in `tests/support.py` and use it here. This way we won't have modify `sys.path`.

File: tests/test_util.py [hunk 4]
```diff
@@ -45,3 +54,15 @@ def test_warn(capsys):
...
+def test_import_app():
+    assert util.import_app("test:app")
+
+    with pytest.raises(ImportError) as e_info:
+        util.import_app("a:app")
+    assert "No module" in str(e_info)
+    
+    with pytest.raises(AppImportError) as e_info:
+        util.import_app("test:wrong_app")
+    assert "Failed to " in str(e_info)
```
Comment (berkerpeksag) [inline tests/test_util.py]: Can we check for `"Failed to find application object 'wrong_app' in 'test'"` instead of `"Failed to"`.

File: tests/test_util.py [hunk 5]
```diff
@@ -45,3 +54,15 @@ def test_warn(capsys):
...
+
+
+def test_import_app():
+    assert util.import_app("test:app")
+
+    with pytest.raises(ImportError) as e_info:
+        util.import_app("a:app")
+    assert "No module" in str(e_info)
+    
+    with pytest.raises(AppImportError) as e_info:
```
Comment (berkerpeksag) [inline tests/test_util.py]: Style nit: `err` is more common name in this file. I'd suggest using it to be consistent with the rest of the file.

**Comment by berkerpeksag** [comment]
@JJ, @lulivi: I see that this PR co-authored by both of you. Do you want us to credit the second author as `Co-Authored-By: Your Full Name <your@email.address>` in the commit message?

File: tests/test_util.py (hunk 4 above)
Comment (JJ) [reply tests/test_util.py]: Sure!

File: tests/test_util.py (hunk 5 above)
Comment (JJ) [reply tests/test_util.py]: OK

File: tests/test_util.py (hunk 3 above)
Comment (JJ) [reply tests/test_util.py]: OK, I'll try to do that.

File: gunicorn/util.py (hunk 2 above)
Comment (JJ) [reply gunicorn/util.py]: OK

File: .gitignore (hunk 1 above)
Comment (JJ) [reply .gitignore]: OK, I'll revert that.

**Comment by lulivi** [comment]
@berkerpeksag: Yes, thank you.

File: tests/test_util.py (hunk 3 above)
Comment (lulivi) [reply tests/test_util.py]: @berkerpeksag You mean to copy-paste [the example app function](https://github.com/benoitc/gunicorn/blob/master/examples/test.py) into `test/support.py`?

Comment (berkerpeksag) [reply tests/test_util.py]: @lulivi yes, that's what I meant.

File: tests/test_util.py [hunk 6]
```diff
@@ -45,3 +46,16 @@ def test_warn(capsys):
...
+    object_app = "support"
+    assert util.import_app(object_app + ":app")
+
+    with pytest.raises(ImportError) as err:
+        util.import_app("a:app")
+    assert "No module" in str(err)
+
+    with pytest.raises(AppImportError) as err:
+        util.import_app(object_app + ":wrong_app")
+    assert "Failed to find application object 'wrong_app' in '{}'".format(object_app) in str(err)
```
Comment (berkerpeksag) [inline tests/test_util.py]: Style nit: Can you move this to a variable:

```py
msg = '...'

assert msg.format(...) in str(err)
```

Comment (berkerpeksag) [inline tests/test_util.py]: `'{}'` -> `{0!r}` (we need to add 0 to make tests pass in Python 2.6)

File: tests/support.py [hunk 7]
```diff
@@ -26,16 +46,20 @@ def wrapper(*args, **kw):
                     if version < min_version:
                         min_version_txt = '.'.join(map(str, min_version))
                         raise unittest.SkipTest(
-                            "Mac OS X %s or higher required, not %s"
-                            % (min_version_txt, version_txt))
```
Comment (berkerpeksag) [inline tests/support.py]: Please revert unrelated cosmetic changes. Unfortunately, they make the diff more noisy and make the review process harder.

File: tests/test_util.py (hunk 6 above)
Comment (lulivi) [reply tests/test_util.py]: Ok!

File: tests/support.py [hunk 8]
```diff
@@ -2,17 +2,37 @@
...
+def app(environ, start_response):
+    """Simplest possible application object"""
+
+    data = b'Hello, World!\n'
+    status = '200 OK'
+
+    response_headers = [
+        ('Content-type', 'text/plain'),
+        ('Content-Length', str(len(data))),
+        ('X-Gunicorn-Version', __version__),
```
Comment (berkerpeksag) [inline tests/support.py]: We can drop the X-Gunicorn-Version header.

File: tests/support.py (hunk 7 above)
Comment (lulivi) [reply tests/support.py]: I'm on it.

File: .gitignore [hunk 9]
```diff
@@ -13,4 +13,4 @@ coverage.xml
 examples/frameworks/pylonstest/PasteScript*
 examples/frameworks/pylonstest/pylonstest.egg-info/
 examples/frameworks/django/testing/testdb.sql
-.tox
+.tox
```
Comment (berkerpeksag) [inline .gitignore]: No new line at end of file.

File: tests/test_util.py [hunk 10]
```diff
@@ -45,3 +46,16 @@ def test_warn(capsys):
     util.warn('test warn')
     _, err = capsys.readouterr()
     assert '!!! WARNING: test warn' in err
+
+
+def test_import_app():
+    object_app = "support"
```
Comment (berkerpeksag) [inline tests/test_util.py]: Style nit: Rest of the file uses sing quotes. It would be nice to be consistent with the rest of the file.

File: tests/test_util.py (hunk 6 above)
Comment (lulivi) [reply tests/test_util.py]: Ok

**Comment by berkerpeksag** [comment]
I've addressed rest of my comments. Let's see what other maintainers say about the change. Thank you!

**Comment by JJ** [comment]
Thanks!

File: tests/support.py (hunk 8 above)
Comment (benoitc) [reply tests/support.py]: also the commit message should be  "remove Cyrillic chars" this has nothing to do with "russian" .

**Comment by benoitc** [comment]
changes look good for me, we should make sure to merge it at one commit.

**Comment by berkerpeksag** [comment]
Squashed commits into one in 6ba2ebe545f3540acb06cc768c74506e13c8901d.

**Comment by berkerpeksag** [comment]
Thanks @JJ and @lulivi!
